dissect/disc/disc.py

```diff
@@ -0,0 +1,756 @@
+from __future__ import annotations
+
+import logging
+from collections import defaultdict
+from datetime import datetime, timedelta, timezone
+from enum import Enum
+from io import BytesIO
+from typing import BinaryIO, Iterator, Optional, Union
+
+from dissect.cstruct import Instance
+from dissect.util.stream import RangeStream
+
+from dissect.disc.c_iso_9660 import c_iso
+from dissect.disc.c_rockridge import (
+    ROCKRIDGE_MAGICS,
+    SUSP_MAGIC,
+    RockRidgeSignature,
+    RockRidgeTimestampType,
+    SystemUseSignature,
+    c_rockridge,
+)
+from dissect.disc.c_udf import UDF_MAGICS
+from dissect.disc.exceptions import (
+    FileNotFoundError,
+    NotADirectoryError,
+    NotAFileError,
+    NotASymlinkError,
+)
+
+logger = logging.getLogger(__name__)
```

Comment:
Can you do this similar to e.g. https://github.com/fox-it/dissect.hypervisor/blob/main/dissect/hypervisor/disk/vmdk.py#L21-L22